### ISSUE TITLE
OpenAPI specifications contains various semantic errors

### ISSUE THREAD

**Issue by maximiliankolb** [DependencyTrack/dependency-track#2492]
### Current Behavior

I am running dependency-track on my local machine. I've downloaded the `swagger.json` file from `http://localhost:8081/api/swagger.json` and inserted it into the swagger-editor as follows: `docker run --rm -d -p 8079:8080 -v $(pwd):/tmp -e SWAGGER_FILE=/tmp/swagger.json swaggerapi/swagger-editor`. Navigating to `http://localhost:8079/` shwos the following error: 

````
Errors

Semantic error at paths./v1/bom.post.parameters
Parameters cannot have both a "in: body" and "in: formData", as "formData" _will_ be the body
Jump to line 534
Semantic error at paths./v1/repository/{uuid}
Equivalent paths are not allowed.
Jump to line 4852
Semantic error at paths./v1/vex.post.parameters
Parameters cannot have both a "in: body" and "in: formData", as "formData" _will_ be the body
Jump to line 6274
Semantic error at definitions.AnalysisRequest.required.0
Read only properties cannot be marked as required by a schema.
Jump to line 7060
Semantic error at definitions.AnalysisRequest.required.1
Read only properties cannot be marked as required by a schema.
Jump to line 7060
Semantic error at definitions.BomSubmitRequest.required.0
Read only properties cannot be marked as required by a schema.
Jump to line 7765
Semantic error at definitions.BomSubmitRequest.required.1
Read only properties cannot be marked as required by a schema.
Jump to line 7765
Semantic error at definitions.CloneProjectRequest.required.0
Read only properties cannot be marked as required by a schema.
Jump to line 9523
Semantic error at definitions.CloneProjectRequest.required.1
Read only properties cannot be marked as required by a schema.
Jump to line 9523
Semantic error at definitions.VexSubmitRequest.required.0
Read only properties cannot be marked as required by a schema.
Jump to line 9615
Semantic error at definitions.VexSubmitRequest.required.1
Read only properties cannot be marked as required by a schema.
Jump to line 9615
Semantic error at definitions.ViolationAnalysisRequest.required.0
Read only properties cannot be marked as required by a schema.
Jump to line 9641
Semantic error at definitions.ViolationAnalysisRequest.required.1
Read only properties cannot be marked as required by a schema.
Jump to line 9641
````

### Steps to Reproduce

1. Download swagger.json
2. Upload swagger.json to swagger-editor container

### Expected Behavior

swagger.json conforms to the OpenAPI specification.

### Dependency-Track Version

4.6.x

### Dependency-Track Distribution

Container Image

### Database Server

PostgreSQL

### Browser

Google Chrome

### Checklist

- [X] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#filing-issues)
- [X] I have checked the [existing issues](https://github.com/DependencyTrack/dependency-track/issues) for whether this defect was already reported

**Comment by maximiliankolb** [comment]
This might be related to https://github.com/DependencyTrack/dependency-track/issues/1276 and https://github.com/DependencyTrack/dependency-track/issues/840.

**Comment by nscuro** [comment]
Thanks for reporting @maximiliankolb! 

And yes, as you already found out, the the OpenAPI spec file needs (re-)work.

I wasn't aware however that the syntax of the existing file is invalid. Did you try to validate the Swagger 2 doc with an OpenAPI 3 validator maybe?

**Comment by maximiliankolb** [comment]
I am unsure. If I use https://apitools.dev/swagger-parser/online/, which according to their repo on [Github](https://github.com/APIDevTools/swagger-parser) parses swagger 2.0 files, then I also get an error as follows:

````
Validation failed. /paths/v1/bom/post has body parameters and formData parameters. Only one or the other is allowed.
````

This points IMHO towards the same issue as mentioned above.

**Comment by nscuro** [comment]
Resolved via #3728. We now have tests that validate the generated OpenAPI spec.
